src/server/views/lists/partials/answers-box.njk

```diff
@@ -1,123 +1,129 @@
 {% import "./macros.njk" as macros %}
 <div class="answers-box">
   <p class="govuk-body">Your answers:</p>
-  {% if not country %}
+   {% if not answers | length %}
     <p class="govuk-body">Your answers will appear here</p>
-  {% endif %}
-  {% if serviceType %}
-    {# until other services are integrated we can hide this #}
-    {# <p class="govuk-body">
-      What are you looking for?<br />
-      <strong>{{ _.upperFirst(_.lowerCase(serviceType)) }}</strong>
-      <a class="govuk-link" href="{{ listsRoutes.finder }}?{{ removeQueryParameter(queryString, 'serviceType') }}">Change</a>
-    </p> #}
-  {% endif %}
-  {% if region %}
+   {% endif %}
+
+  {% set region = answers.region if answers.region else "Not supplied" %}
+  {% if answers.region or answers.region === "" or isResultsPage %}
+  {% set returnParam = "?return=results" if isResultsPage else "" %}
     <p class="govuk-body">
-      Location<br />
-      <strong>{{ _.upperFirst(region) }}</strong>{% if print !== "yes" %}
-      <a class="govuk-link noprint" href="{{ listsRoutes.finder }}?{{ removeQueryParameter(queryString, 'region') }}" aria-label="Change country's area answer">Change</a>{% endif %}
+      <span id="location">Location<br /></span>
+      <strong aria-labelledby="location">{{ _.upperFirst(answers.region) if answers.region else "Not supplied" }}</strong>{% if print !== "yes" %}
+      <a class="govuk-link noprint" href="/find/{{ serviceType }}/{{ answers.country }}/region{{ returnParam }}" aria-label="Change country's area answer">
+        Change
+      </a>
+      {% endif %}
     </p>
   {% endif %}
-  {% if country %}
+
+  {% if answers.country or isResultsPage %}
+    {% set countryAnswer = answers.country %}
     <p class="govuk-body">
-      Country<br />
-      <strong>{{ country }}</strong>{% if print !== "yes" %}
-      <a class="govuk-link noprint" href="{{ listsRoutes.finder }}?{{ removeQueryParameter(queryString, 'country') }}" aria-label="Change country answer">Change</a>{% endif %}
+      <span id="country">Country<br /></span>
+      <strong aria-labelledby="country">{{ countryAnswer }}</strong>{% if print !== "yes" %}
+      <a class="govuk-link noprint" href="/find/{{ serviceType }}/country{{ returnParam }}" aria-label="Change country answer">Change</a>{% endif %}
```

Comment:
I'm sure you've tested and it maybe obvious in the controller when I get there but does this handle special characters and spaces in the country name?  Same for region.

Reply:
You mean `countryAnswer`? This just renders what was inserted into session. 

I assume you're thinking about URLs specifically where it might be more relevant? Node has two ways of encoding and decoding URLs. I had left some comments on my own PR to describe my change. URLSearchParams and querystring. 

seeing as express uses querystring, then we may as well querystring. I also don't quite like the URLSearchParams API. It is able to parse arrays, however using `.toString()` (which returns the encoded parameters), changes arrays to strings e.g. "a,b,c,d" - then encodes them. It means you have to "split(',')" on the next request.

Reply:
I've a vague recollection that there was a discrepancy between the country name used for the list and the list of countries in the meta data.  As long as all countries are consistent then I think it should be safe to remove, just needs some testing to validate.  Good to know psql has unaccent which should further resolve the headache.